### PR TITLE
feat: ts fuel all events deployment scripts for e2e testing

### PR DESCRIPTION
**Pull request by DenhamPreen**

### CODE REVIEW

**Comment by JonoPrest** [comment]
Hey @DenhamPreen, will need some comments and explanations please 🙏🏼

File: scenarios/fuel_test/contracts/interaction-tools/.env.example
Comment (DenhamPreen) [inline scenarios/fuel_test/contracts/interaction-tools/.env.example:1]: env file for the seed phrase and conf for the network

File: scenarios/fuel_test/contracts/interaction-tools/src/main.rs
Comment (DenhamPreen) [inline scenarios/fuel_test/contracts/interaction-tools/src/main.rs:27]: refactor to take the env vars

File: scenarios/fuel_test/contracts/interaction-tools/src/main.rs
Comment (DenhamPreen) [inline scenarios/fuel_test/contracts/interaction-tools/src/main.rs:38]: This is commented out, same as below as it's clearly left from running the script at some point.

File: scenarios/fuel_test/contracts/ts-interaction-tools/src/contract/AllEventsFactory.ts
Comment (DenhamPreen) [inline scenarios/fuel_test/contracts/ts-interaction-tools/src/contract/AllEventsFactory.ts:1]: All these src/contract files are autogened from the contract abi, cmd is in the readme to gen

File: scenarios/fuel_test/contracts/ts-interaction-tools/src/deploy.ts
Comment (DenhamPreen) [inline scenarios/fuel_test/contracts/ts-interaction-tools/src/deploy.ts:1]: typescript deployment script

**Comment by DenhamPreen** [comment]
PR made as the rust clients don't work on a certain network

File: scenarios/fuel_test/contracts/interaction-tools/src/main.rs
Comment (JonoPrest) [reply scenarios/fuel_test/contracts/interaction-tools/src/main.rs:38]: Should we just remove it if it's not useful?

File: scenarios/fuel_test/contracts/interaction-tools/src/main.rs
Comment (JonoPrest) [inline scenarios/fuel_test/contracts/interaction-tools/src/main.rs:81]: Looks like the println! below will get the wrong intended value for the variable `r` since it was being shadowed

File: scenarios/fuel_test/contracts/ts-interaction-tools/src/contract/AllEventsFactory.ts
Comment (JonoPrest) [reply scenarios/fuel_test/contracts/ts-interaction-tools/src/contract/AllEventsFactory.ts:1]: Should we hide them from git if they are gitignored?

**Review by JonoPrest — APPROVED** [review]
Cool @DenhamPreen, 

Could you just get back to the comments/questions before merge?

I'm not sure if we should if we should be committing generated code. 

Tagging @DZakh here for when he gets back.

**Comment by DenhamPreen** [comment]
Thanks for your feedback @JonoPrest, I'll address and get back to this when I have a chance

File: scenarios/fuel_test/contracts/interaction-tools/src/main.rs
Comment (DenhamPreen) [reply scenarios/fuel_test/contracts/interaction-tools/src/main.rs:38]: removed, it's in the git history if @DZakh you still need it

**Comment by DenhamPreen** [comment]
Hmm.. Something went weird here with the rebase. I'm not sure why this happens but have seen it before.

**Comment by DenhamPreen** [comment]
I don't understand how this just merged. I did not merge it

**Pull request merged**
